src/Metadata/PropertyTypeArray.php

```diff
@@ -19,29 +19,29 @@ final class PropertyTypeArray extends AbstractPropertyType
     private $hashmap;
 
     /**
-     * @var bool
+     * @var string
      */
-    private $isCollection;
+    private $collectionClass;
 
-    public function __construct(PropertyType $subType, bool $hashmap, bool $nullable, bool $isCollection = false)
+    public function __construct(PropertyType $subType, bool $hashmap, bool $nullable, ?string $collectionClass = null)
```

Comment:
if somebody implemented their own parser that creates this class, this is a BC break.

i don't think we can do this in a backwards compatible way without making things very complicated (we could allow bool and then fallback to the old behaviour, but that would make things a lot more convoluted)

i think we should start a new major version for this.

Reply:
keeping the signature BC is doable, but the merging would then not be possible and the code would become quite a bit more complicated to handle both cases. if you think its easy to do it in BC way, please do but otherwise we can start that 2.x branch and we can create a milestone with issues for things that we want to change for 2.0 to coordinate when we can release.